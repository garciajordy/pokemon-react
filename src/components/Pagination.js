import React from 'react';
import PropTypes from 'prop-types';

function Pagination({ goToPrevPage, goToNextPage }) {
  return (
    <div className="button-bottom">
      {goToPrevPage ? <button type="button" onClick={goToPrevPage}>Previous</button> : 'First Page'}
      {goToNextPage ? <button type="button" onClick={goToNextPage}>Next</button> : 'Last Page'}
    </div>
  );
}

Pagination.propTypes = {
  goToPrevPage: PropTypes.func,
  goToNextPage: PropTypes.func,
};

Pagination.defaultProps = {
  goToNextPage: null,
  goToPrevPage: null,
};

export default Pagination;
