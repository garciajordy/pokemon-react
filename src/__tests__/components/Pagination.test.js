import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import Pagination from '../../components/Pagination';

afterEach(cleanup);
function test() {
  return 5 + 6;
}
it('should take a snapshot', () => {
  const { asFragment } = render(<Pagination
    goToNextPage={test}
    goToPrevPage={test}
  />);
  expect(asFragment(<Pagination
    goToNextPage={test}
    goToPrevPage={test}
  />)).toMatchSnapshot();
});

it('displays heading', () => {
  render(<Pagination
    goToNextPage={test}
    goToPrevPage={test}
  />);
  expect(screen.getByText('Next')).toBeTruthy();
});
