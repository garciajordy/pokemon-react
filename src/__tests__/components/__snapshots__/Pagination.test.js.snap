// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="button-bottom"
  >
    <button
      type="button"
    >
      Previous
    </button>
    <button
      type="button"
    >
      Next
    </button>
  </div>
</DocumentFragment>
`;
